test/response.js

```diff
@@ -1290,6 +1290,37 @@ describe('Response', () => {
             const res = await server.inject('/');
             expect(res.statusCode).to.equal(500);
         });
+
+        it('is only called once for returned responses', async () => {
```

Comment:
so i only see one added test here, and while it does demonstrate nicely that it fixes the original issue, you commented

> This also caught another error, where the marshal() method could sometimes be called after close().

is this something that we can also add a test for? were you able to determine the conditions that caused them to be called out of order? it would be nice to ensure this doesn't regress

Reply:
Thanks for the concern. The tests already include a case that exercise this.

If you remove the `if` from here, you will see that it fails [this](https://github.com/hapijs/hapi/blob/adf341f25996393e6d3bc557203687b29f10281c/test/transmit.js#L526) test: https://github.com/hapijs/hapi/blob/adf341f25996393e6d3bc557203687b29f10281c/lib/transmit.js#L38-L45

Reply:
> The tests already include a case that exercise this.

so the original behavior with that test was running the functions out of order but not crashing because there were no assertions. introducing these assertions that things are run in order caused that test to result in a crash, and the if statement you added there fixes the problem so instead of running them out of order we simply don't run `marshal()` at all.

did i understand that right?

Reply:
Yes, that is exactly it.